screencastr.thor

```diff
@@ -16,49 +16,59 @@ class Screencastr < Thor
   # 6. Upload video to S3
     # a. Will need a naming scheme to match the location properly
 
+  class_option :framerate, aliases: "-f", desc: "Specify a framerate for OUT_FILE, (eg. -f 30)", type: :numeric
+  class_option :resolution, aliases: "-r", desc: "Specify a resolution for OUT_FILE, (eg. -r 1920x1080)", type: :string
+
   desc "add_bumpers IN_FILE OUT_FILE", "Add bumpers to video"
-  method_option :framerate, aliases: "-f", desc: "Specify a framerate for OUT_FILE, (eg. -f 30)", type: :numeric
-  method_option :resolution, aliases: "-r", desc: "Specify a resolution for OUT_FILE, (eg. -r 1920x1080)"
   def add_bumpers(in_file, out_file)
     ProcessHelpers.resolution_require_value(options[:resolution])
-    video = FFMPEG::Movie.new(in_file)
+    ext = File.extname(out_file)[1..-1]
+
+    `ffmpeg -f lavfi -i anullsrc=channel_layout=stereo:sample_rate=44100 \
+    -loop 1 -i assets/1080-GA-Splash.png -f #{ext} -t 5 -r #{options[:framerate] || 30} -pix_fmt yuv420p \
+    -vf scale=#{options[:resolution] || "1920x1080"} -map 0:a -map 1:v bumper.#{ext}`
+
+    # Can't use invoke, because Thor won't allow the same task to be invoked twice
+    begin
+      s = Screencastr.new
+      s.options = options
+      s.concat("bumper.#{ext}", in_file, "bumper-tmp.#{ext}")
+      s.concat("bumper-tmp.#{ext}", "bumper.#{ext}", out_file)
+    ensure
+      File.delete("bumper.#{ext}") if File.exists?("bumper.#{ext}")
+      File.delete("bumper-tmp.#{ext}") if File.exists?("bumper-tmp.#{ext}")
+    end
```

Comment:
So here I'm just reusing the `concat` task from below to concatenate the bumper video with the screencast, once at the beginning of the screencast, once at the end.

Slightly annoying thing about Thor though: you can't use `invoke` to execute the same task more than once, by design. So I worked around this limitation by just calling the method twice directly.

Reply:
Also the `begin..ensure` block is in case something goes wrong with either `concat` call. Want to ensure I remove the temporary bumper videos.

Reply:
Two thoughts here:

1. Can't you pass more than two files to concat, potentially? That would eliminate the need for two steps for the bumpers. (i.e. in one step have `bumper.mp4 input.mp4 bumper.mp4` as the input files)
2. I'm sure it's pretty quick to create the bumper video since it's like 5s and no audio, but perhaps we can create it once and reuse the file if it already exists on the file system, rather than creating it and deleting it every time.

Thoughts?

Reply:
1. The only issue I foresee with concatenating many files at once is that it'll change the `filter_complex` pretty significantly depending on how many files are being concatenated. Definitely doable though.
2. Yeah, we can definitely do that, just to save a couple seconds each run.